src/primitives/block.cpp

```diff
@@ -207,12 +206,29 @@ bool CBlock::SignBlock(const CKeyStore& keystore)
             if (!keystore.GetKey(keyID, key))
                 return false;
 
-            //vector<unsigned char> vchSig;
             if (!key.Sign(GetHash(), vchBlockSig))
                  return false;
 
             return true;
         }
+        else if(whichType == TX_WITNESS_V0_SCRIPTHASH || whichType == TX_WITNESS_V0_KEYHASH)
```

Comment:
indentation here

Reply:
script hash addresses won't work. they require a larger script sig including the transaction data.